src/overlays/actors/ovl_En_Takaraya/z_en_takaraya.c

```diff
@@ -28,61 +47,360 @@ ActorInit En_Takaraya_InitVars = {
+        if (play->msgCtx.choiceIndex == 0) {
+            if (gSaveContext.save.saveInfo.playerData.rupees < play->msgCtx.unk1206C) {
+                this->actor.textId = 0x77B;
+                if (this->skelAnime.animation == &object_bg_Anim_009890) {
+                    Animation_MorphToPlayOnce(&this->skelAnime, &object_bg_Anim_000968, 5.0f);
+                }
+                Audio_PlaySfx(NA_SE_SY_ERROR);
+            } else {
+                Audio_PlaySfx_MessageDecide();
+                Rupees_ChangeBy(play->msgCtx.unk1206C * -1);
+                func_80ADEF74(this, play);
+                this->actor.textId = 0x778;
+                if (this->skelAnime.animation != &object_bg_Anim_009890) {
+                    Animation_MorphToLoop(&this->skelAnime, &object_bg_Anim_009890, 5.0f);
+                }
+            }
+        } else {
+            Audio_PlaySfx_MessageCancel();
+            this->actor.textId = D_80ADFB44[(void)0, gSaveContext.save.playerForm];
+            if (this->skelAnime.animation == &object_bg_Anim_009890) {
+                Animation_MorphToPlayOnce(&this->skelAnime, &object_bg_Anim_000968, 5.0f);
+            }
+        }
+        Message_ContinueTextbox(play, this->actor.textId);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/EnTakaraya_Init.s")
+void func_80ADF2D4(EnTakaraya* this) {
+    Actor_PlaySfx_FlaggedCentered2(&this->actor, NA_SE_SY_FOUND);
+    Audio_PlaySubBgm(NA_BGM_TIMED_MINI_GAME);
+    this->timer = 145;
+    SET_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_WAIT);
+    CLEAR_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_TIME_PASSED);
+    this->actionFunc = &func_80ADF338;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/EnTakaraya_Destroy.s")
+void func_80ADF338(EnTakaraya* this, PlayState* play) {
+    Actor* chest = play->actorCtx.actorLists[ACTORCAT_CHEST].first;
+    Actor* player = play->actorCtx.actorLists[ACTORCAT_PLAYER].first;
+    Vec3f eye;
+    Vec3f at;
+    f32 sp2C;
+    s16 CamId;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADED8C.s")
+    SkelAnime_Update(&this->skelAnime);
+    if ((CutsceneManager_GetCurrentCsId() == this->actor.csId) && (chest != NULL)) {
+        this->timer--;
+        CamId = CutsceneManager_GetCurrentSubCamId(this->actor.csId);
+        if (this->timer >= 26) {
+            sp2C = 250.0f;
+        } else {
+            sp2C = ((chest->xzDistToPlayer - 250.0f) * (25 - this->timer) * 0.04f) + 250.0f;
+        }
+        eye.x = (Math_SinS(chest->yawTowardsPlayer) * sp2C) + chest->world.pos.x;
+        eye.y = player->world.pos.y + 120.0f;
+        eye.z = (Math_CosS(chest->yawTowardsPlayer) * sp2C) + chest->world.pos.z;
+        at.x = eye.x - (Math_SinS(chest->yawTowardsPlayer) * 250.0f);
+        at.y = eye.y - 90.0f;
+        at.z = eye.z - (Math_CosS(chest->yawTowardsPlayer) * 250.0f);
+        Play_SetCameraAtEye(play, CamId, &at, &eye);
```

Comment:
```suggestion
        Play_SetCameraAtEye(play, CamId, &subCamAt, &subCamEye);
```
Variables can be called like this